source/backend/api/Services/PropertyService.cs

```diff
@@ -377,6 +377,40 @@ public void UpdateLocation(PimsProperty incomingProperty, ref PimsProperty prope
             }
         }
 
+        public T PopulateNewFileProperty<T>(T fileProperty)
+        {
+            // TODO: Remove this casting when LOCATION gets added to all remaining file-property types (research, disposition, lease)
+            if (fileProperty is PimsPropertyAcquisitionFile acquisitionFileProperty)
+            {
+                // convert spatial location from lat/long (4326) to BC Albers (3005) for database storage
+                var geom = acquisitionFileProperty.Location;
+                if (geom is not null && geom.SRID != SpatialReference.BCALBERS)
+                {
+                    var newCoords = _coordinateService.TransformCoordinates(geom.SRID, SpatialReference.BCALBERS, geom.Coordinate);
+                    acquisitionFileProperty.Location = GeometryHelper.CreatePoint(newCoords, SpatialReference.BCALBERS);
+                }
+            }
+
+            return fileProperty;
+        }
+
+        public void UpdateFilePropertyLocation<T>(T incomingFileProperty, T filePropertyToUpdate)
+            where T : IWithPropertyEntity
+        {
+            // TODO: Remove this casting when LOCATION gets added to all remaining file-property types (research, disposition, lease)
+            if (incomingFileProperty is PimsPropertyAcquisitionFile incomingAcquisitionProperty
+                && filePropertyToUpdate is PimsPropertyAcquisitionFile acquisitionPropertyToUpdate)
+            {
+                // convert spatial location from lat/long (4326) to BC Albers (3005) for database storage
+                var geom = incomingAcquisitionProperty.Location;
```

Comment:
does the above casting also check for null?

Reply:
According to MS docs: https://learn.microsoft.com/en-us/dotnet/csharp/fundamentals/tutorials/safely-cast-using-pattern-matching-is-and-as-operators
![image](https://github.com/bcgov/PSP/assets/24322224/d5e076eb-22b9-4c24-9114-a0247bfa80c8)

Reply:
Also - a note that this casting will go away this sprint with the story to use location on other file types